temporal-sdk/src/main/java/io/temporal/client/WorkflowStartOperationUpdate.java

```diff
@@ -0,0 +1,258 @@
+
+  public static <T, R> Builder<R> newBuilder(Functions.Func1<T, R> update) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {});
+  }
+
+  public static <T, A1, R> Builder<R> newBuilder(Functions.Func2<T, A1, R> update, A1 arg1) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1});
+  }
+
+  public static <T, A1, A2, R> Builder<R> newBuilder(
+      Functions.Func3<T, A1, A2, R> update, A1 arg1, A2 arg2) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2});
+  }
+
+  public static <T, A1, A2, A3, R> Builder<R> newBuilder(
+      Functions.Func4<T, A1, A2, A3, R> update, A1 arg1, A2 arg2, A3 arg3) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2, arg3});
+  }
+
+  public static <T, A1, A2, A3, A4, R> Builder<R> newBuilder(
+      Functions.Func5<T, A1, A2, A3, A4, R> update, A1 arg1, A2 arg2, A3 arg3, A4 arg4) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4});
+  }
+
+  public static <T, A1, A2, A3, A4, A5, R> Builder<R> newBuilder(
+      Functions.Func6<T, A1, A2, A3, A4, A5, R> update,
+      A1 arg1,
+      A2 arg2,
+      A3 arg3,
+      A4 arg4,
+      A5 arg5) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4, arg5});
+  }
+
+  public static <T> Builder<Void> newBuilder(Functions.Proc1<T> update) {
+    SerializedLambda meta = JavaLambdaUtils.toSerializedLambda(update);
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {});
+  }
+
+  public static <T, A1> Builder<Void> newBuilder(Functions.Proc2<T, A1> update, A1 arg1) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1});
+  }
+
+  public static <T, A1, A2> Builder<Void> newBuilder(
+      Functions.Proc3<T, A1, A2> update, A1 arg1, A2 arg2) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2});
+  }
+
+  public static <T, A1, A2, A3> Builder<Void> newBuilder(
+      Functions.Proc4<T, A1, A2, A3> update, A1 arg1, A2 arg2, A3 arg3) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2, arg3});
+  }
+
+  public static <T, A1, A2, A3, A4> Builder<Void> newBuilder(
+      Functions.Proc5<T, A1, A2, A3, A4> update, A1 arg1, A2 arg2, A3 arg3, A4 arg4) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4});
+  }
+
+  public static <T, A1, A2, A3, A4, A5> Builder<Void> newBuilder(
+      Functions.Proc6<T, A1, A2, A3, A4, A5> update, A1 arg1, A2 arg2, A3 arg3, A4 arg4, A5 arg5) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4, arg5});
+  }
+
+  public static <R> Builder<R> newBuilder(String update, Class<R> resultType, Object[] args) {
```

Comment:
Other builders allow you to create an empty builder and set everything later. There are some ways we can do this.

For the most part builders accept simple values without much logic. So a builder for such a thing here might accept an update name + result type tuple (really just two fields with mutual exclusivity checks) or an update method reference `Object`.

For example:

```java
public static class Builder<R> {
  // Can only either have the first two or the third
  private String updateName;
  private Class<R> updateResultClass;
  private Object updateMethodReference;

  private Object[] updateArgs;
  private String updateId;
  private WorkflowUpdateStage waitForStage;

  private Builder() { }

  private Builder(WorkflowStartOperationUpdate existing) {
    // Copy all 6 fields because they are on that update object verbatim
  }

  public Builder<R> setUpdate(String updateName, Class<R> updateResultClass) {
    this.updateName = updateName;
    this.updateResultClass = updateResultClass;
    updateMethodReference = null;
    return this;
  }

  public Builder<R> setUpdate(Object updateMethodReference) {
    this.updateMethodReference = updateMethodReference;
    updateName = null;
    updateResultClass = null;
  }

  // Setters for the other three fields
}
```

You can/should still have all of your `newBuilder` helpers, but I think this is a vastly simpler approach and allows you to match our builder pattern (can create empty, can create from existing object, etc), and allows you to late-bind the validation. Also it removes the `UpdateOptions` which, if we're not going to expose as is, there's no need building until needed internally.

Reply:
I wasn't 100% sure if the new builder fulfills all the suggestions here; so I left this comment unresolved.

re `UpdateOptions`: I'm inclined to re-use it here; for example, because it allows users to obtain the updateId before executing the request via `.getOptions().getUpdateId()`.

Reply:
I'll resolve it, this applied to a previous form of this PR.